File: src/main/java/com/viartemev/requestmapper/annotations/UnknownAnnotation.java
package com.viartemev.requestmapper.annotations;

import com.viartemev.requestmapper.RequestMappingItem;

import java.util.Collections;
import java.util.List;

public class UnknownAnnotation implements MappingAnnotation {

    private static UnknownAnnotation unknownAnnotation = new UnknownAnnotation();

    private UnknownAnnotation() {

    }

    public static UnknownAnnotation getInstance() {
        return unknownAnnotation;
    }

    @Override
    public List<RequestMappingItem> values() {
        return Collections.emptyList();
    }

}

File: src/main/kotlin/com/viartemev/requestmapper/annotations/UnknownAnnotation.kt
package com.viartemev.requestmapper.annotations

import com.viartemev.requestmapper.RequestMappingItem

object UnknownAnnotation : MappingAnnotation {
    private val mappingItems = emptyList<RequestMappingItem>()
    override fun values(): List<RequestMappingItem> = mappingItems
}
